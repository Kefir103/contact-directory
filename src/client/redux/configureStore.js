import { createStore, applyMiddleware, compose } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/index';
import { rootReducer } from './reducers/rootReducer';
import thunk from 'redux-thunk';

let store;

export default function configureStore(initialState) {
    if (store) {
        return store;
    }

    store =
        process.env.NODE_ENV === 'production'
            ? createStore(rootReducer, initialState, compose(applyMiddleware(thunk)))
            : createStore(rootReducer, initialState, composeWithDevTools(applyMiddleware(thunk)));

    return store;
}
