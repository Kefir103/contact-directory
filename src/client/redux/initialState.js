export const initialState = {
    filter: {
        sortingMap: new Map([]),
        filterText: '',
        filterFields: [],
        currentPage: 1,
    },
    data: {
        currentElement: {
            elementInfo: undefined,
            elementIndex: -1,
        },
        elements: [],
        currentElements: [],
        filteredElements: undefined,
        inputContainer: {
            isAddButtonDisabled: false,
            isFormOpen: false,
            inputElement: {
                id: 0,
                firstName: '',
                lastName: '',
                email: '',
                phone: '',
            },
            validInputs: {
                id: false,
                firstName: false,
                lastName: false,
                email: false,
                phone: false,
            },
            addingStatus: {
                addingStatusText: '',
                isError: false,
            },
        },
    },
    appStatus: {
        isLoading: false,
        error: undefined,
    },
};
