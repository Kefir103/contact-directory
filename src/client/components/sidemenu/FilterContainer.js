import React from 'react';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch } from '@fortawesome/free-solid-svg-icons';
import { bindActionCreators } from 'redux';
import { setFilterFields, setFilterText } from '../../redux/actions/filterActions';
import { filterElements } from '../../redux/actions/dataActions';

const FilterContainer = (props) => {
    const handleFilterFormSubmit = (event) => {
        event.preventDefault();
        if (props.filterText) {
            props.actions.filterElements(props.elements, props.filterText, props.filterFields);
        }
    };

    const handleFilterTextChanged = (event) => {
        event.preventDefault();
        props.actions.setFilterText(event.target.value);
        if (!event.target.value) {
            props.actions.filterElements();
        }
    };

    const handleFilterValueChecked = (event) => {
        if (event.target.checked) {
            props.actions.setFilterFields([...props.filterFields, event.target.name]);
        } else {
            const deletingElementIndex = props.filterFields.findIndex(
                (field) => field === event.target.name
            );
            const newFilterFields = [...props.filterFields];
            newFilterFields.splice(deletingElementIndex, 1);
            props.actions.setFilterFields(newFilterFields);
        }
    };

    return (
        <div className={'filter-container'}>
            <p>Фильтрация данных</p>
            <form id={'filter-form'} onSubmit={handleFilterFormSubmit}>
                <input
                    type={'search'}
                    placeholder={'Введите данные'}
                    value={props.filterText}
                    onChange={handleFilterTextChanged}
                />
                <button type={'submit'} disabled={!(props.filterText && props.filterFields.length)}>
                    <FontAwesomeIcon icon={faSearch} size={'1x'} />
                </button>
                <label>
                    <input type={'checkbox'} name={'id'} onChange={handleFilterValueChecked} />
                    Id
                </label>
                <label>
                    <input
                        type={'checkbox'}
                        name={'firstName'}
                        onChange={handleFilterValueChecked}
                    />
                    First Name
                </label>
                <label>
                    <input
                        type={'checkbox'}
                        name={'lastName'}
                        onChange={handleFilterValueChecked}
                    />
                    Last Name
                </label>
                <label>
                    <input type={'checkbox'} name={'email'} onChange={handleFilterValueChecked} />
                    E-Mail
                </label>
                <label>
                    <input type={'checkbox'} name={'phone'} onChange={handleFilterValueChecked} />
                    Phone
                </label>
            </form>
        </div>
    );
};

const mapStateToProps = (state) => {
    return {
        filterFields: state.filter.filterFields,
        filterText: state.filter.filterText,
        elements: state.data.elements,
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        actions: bindActionCreators(
            {
                setFilterFields,
                setFilterText,
                filterElements,
            },
            dispatch
        ),
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(FilterContainer);
