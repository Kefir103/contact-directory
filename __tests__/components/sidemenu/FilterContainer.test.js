import React from 'react';
import configureMockStore from 'redux-mock-store';
import { shallow, render } from 'enzyme';
import FilterContainer from '../../../src/client/components/sidemenu/FilterContainer';
import { initialState } from '../../../src/client/redux/initialState';

const mockStore = configureMockStore();

describe('<FilterContainer />', () => {
    test('should render correctly without filterText', () => {
        const store = mockStore({ ...initialState });
        const wrapper = shallow(<FilterContainer store={store} />);
        const component = render(wrapper);
        expect(component).toMatchSnapshot();
    });
    test('should render correctly with filterText', () => {
        const store = mockStore({
            ...initialState,
            filter: {
                ...initialState.filter,
                filterText: 'filterText',
            },
        });
        const wrapper = shallow(<FilterContainer store={store} />);
        const component = render(wrapper);
        expect(component).toMatchSnapshot();
    });
    test('should render correctly with filterText and checked filterFields', () => {
        const store = mockStore({
            ...initialState,
            filter: {
                ...initialState.filter,
                filterText: 'filterText',
                filterFields: ['id', 'firstName', 'lastName', 'email', 'phone'],
            },
        });
        const wrapper = shallow(<FilterContainer store={store} />);
        const component = render(wrapper);
        expect(component).toMatchSnapshot();
    });
});
