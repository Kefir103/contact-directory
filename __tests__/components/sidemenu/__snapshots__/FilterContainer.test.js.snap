// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterContainer /> should render correctly with filterText 1`] = `
<div
  class="filter-container"
>
  <p>
    Фильтрация данных
  </p>
  <form
    id="filter-form"
  >
    <input
      placeholder="Введите данные"
      type="search"
      value="filterText"
    />
    <button
      disabled=""
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-search fa-w-16 fa-1x "
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
        />
      </svg>
    </button>
    <label>
      <input
        name="id"
        type="checkbox"
      />
      Id
    </label>
    <label>
      <input
        name="firstName"
        type="checkbox"
      />
      First Name
    </label>
    <label>
      <input
        name="lastName"
        type="checkbox"
      />
      Last Name
    </label>
    <label>
      <input
        name="email"
        type="checkbox"
      />
      E-Mail
    </label>
    <label>
      <input
        name="phone"
        type="checkbox"
      />
      Phone
    </label>
  </form>
</div>
`;

exports[`<FilterContainer /> should render correctly with filterText and checked filterFields 1`] = `
<div
  class="filter-container"
>
  <p>
    Фильтрация данных
  </p>
  <form
    id="filter-form"
  >
    <input
      placeholder="Введите данные"
      type="search"
      value="filterText"
    />
    <button
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-search fa-w-16 fa-1x "
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
        />
      </svg>
    </button>
    <label>
      <input
        name="id"
        type="checkbox"
      />
      Id
    </label>
    <label>
      <input
        name="firstName"
        type="checkbox"
      />
      First Name
    </label>
    <label>
      <input
        name="lastName"
        type="checkbox"
      />
      Last Name
    </label>
    <label>
      <input
        name="email"
        type="checkbox"
      />
      E-Mail
    </label>
    <label>
      <input
        name="phone"
        type="checkbox"
      />
      Phone
    </label>
  </form>
</div>
`;

exports[`<FilterContainer /> should render correctly without filterText 1`] = `
<div
  class="filter-container"
>
  <p>
    Фильтрация данных
  </p>
  <form
    id="filter-form"
  >
    <input
      placeholder="Введите данные"
      type="search"
      value=""
    />
    <button
      disabled=""
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-search fa-w-16 fa-1x "
        data-icon="search"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          fill="currentColor"
        />
      </svg>
    </button>
    <label>
      <input
        disabled=""
        name="id"
        type="checkbox"
      />
      Id
    </label>
    <label>
      <input
        disabled=""
        name="firstName"
        type="checkbox"
      />
      First Name
    </label>
    <label>
      <input
        disabled=""
        name="lastName"
        type="checkbox"
      />
      Last Name
    </label>
    <label>
      <input
        disabled=""
        name="email"
        type="checkbox"
      />
      E-Mail
    </label>
    <label>
      <input
        disabled=""
        name="phone"
        type="checkbox"
      />
      Phone
    </label>
  </form>
</div>
`;
